biojava-core/src/test/java/org/biojava/nbio/core/sequence/io/GenbankReaderTest.java

```diff
@@ -267,6 +294,43 @@ public void testNcbiExpandedAccessionFormats() throws Exception {
 		DNASequence header2 = readGenbankResource("/empty_header2.gb");
 		assertEquals("AZZZAA02123456789 10000000000 bp    DNA     linear   PRI 15-OCT-2018", header2.getOriginalHeader());
 	}
+	
+	@Test
+	public void testLegacyLocusCompatable() throws Exception {
+		
+		// Testing opening a genbank file with uppercase units, strand and topology
+		DNASequence header0 = readGenbankResource("/uppercase_locus0.gb");
+		assertEquals("ABC12.3_DE   7071 BP DS-DNA   CIRCULAR  SYN       22-JUL-1994", header0.getOriginalHeader());
```

Comment:
Could you add a few extra assertions, to be sure that the name, accession, and compoundType are still successfully pulled out from the regex pattern groups?

Reply:
I will do

Reply:
I have had a look into this and found that I can check the ID in the locus line if I remove the ACCESSION line from the header in the test genbank files as the locus line ID is used as a default accession if ACCESSION is missing. 

The name set in the headerParser is subsequently available in the resulting sequence object.

The only other 2 regex groups that are used in GenbankSequenceParser are the units and molecule type which are both used to set the private CompundSet object compoundType.

I can not however find how compoundType is subsequently used to define the CompoundSet of sequence created by GenbankReader.

To me it appears that it is only possible to read a genbank using GenbankReader when you already know the molecule type the file contains. As the only classes that implement the SequenceCreatorInterface which is required for creating a GenbankReader object are:

- CasePreservingProteinSequenceCreator
- DNASequenceCreator
- FileProxyDNASequenceCreator
- FileProxyProteinSequenceCreator
- ProteinSequenceCreator
- RNASequenceCreator

Reply:
Please let me know if I am wrong and it is possible to create sequence object where the CompoundSet is defined by the genbank locus line and I will use that method to test the locus line parsing.

Or if there is another way to test that the correct molecule type is pulled out of the header line.

Thanks,
James

Reply:
> Please let me know if I am wrong and it is possible to create sequence object where the CompoundSet is defined by the genbank locus line and I will use that method to test the locus line parsing.

Sorry, I don't know this version of the Genbank parser that well, I was simply looking at the block starting from line 171

https://github.com/biojava/biojava/pull/835/files#diff-6e7c9bf25ddc5b786dd48e78a10c0052R171

Is `compoundType` set here not used later?

Reply:
From my limited experience with this code I can't see anywhere that uses `compoundType`.

I would appear the the molecule type of the sequence objects returned by `GenbankReader` is determined by the `SequenceCreatorInterface` implementation passed in not the information in the genbank file.

Reply:
It would be interesting to hear from someone with better knowledge of this code that could hopefully clarify this.

Reply:
I think I have a solution, I have added a method to the test that will return an `AbstractSequence` that will have its `CompoundSet` defined by the `getCompoundType()` method from `GenbankSequenceParser`. I have also modified one of the test files to be an RNA sequence.